src/block_cache/block_cache.c

```diff
@@ -374,38 +359,34 @@ __wt_blkcache_get(WT_SESSION_IMPL *session, WT_ITEM *data, const uint8_t *addr,
     TAILQ_FOREACH (blkcache_item, &blkcache->hash[bucket], hashq) {
         if (blkcache_item->addr_size == addr_size && blkcache_item->fid == S2BT(session)->id &&
           memcmp(blkcache_item->addr, addr, addr_size) == 0) {
-            ret = __wt_buf_set(session, data, blkcache_item->data, blkcache_item->data_size);
-            if (ret == 0) {
-                blkcache_item->num_references++;
-                if (blkcache_item->freq_rec_counter < 0)
-                    blkcache_item->freq_rec_counter = 0;
-                blkcache_item->freq_rec_counter++;
-
-                *foundp = true;
-            }
+            blkcache_item->num_references++;
+            if (blkcache_item->freq_rec_counter < 0)
+                blkcache_item->freq_rec_counter = 0;
+            blkcache_item->freq_rec_counter++;
+            blkcache_item->ref_count++;
             break;
         }
     }
     __wt_spin_unlock(session, &blkcache->hash_locks[bucket]);
-    WT_RET(ret);
 
-    if (*foundp) {
+    if (blkcache_item != NULL) {
+        *blkcache_retp = blkcache_item;
+        *foundp = *skip_cachep = true;
```

Comment:
Why are we setting *skip_cachep to true here? On line 351 we are setting *skip_cachep to true when we decide to bypass the cache. Here we are returning a pointer to a cached item, not skipping the cache, and yet *skip_cachep is also set to "true".

Reply:
> Why are we setting *skip_cachep to true here? On line 351 we are setting *skip_cachep to true when we decide to bypass the cache. Here we are returning a pointer to a cached item, not skipping the cache, and yet *skip_cachep is also set to "true".

My understanding of `skip_cache` is that it means "after you read this block, don't call `__wt_blkcache_put()` to store it in the cache". If that is correct, then if we find the block in the cache, it's reasonable to set `skip_cache = true`, because we're not subsequently calling `__wt_blkcache_put()` to store the block in the cache, it's already there.

This simplifies the caller a bit, because it means we're setting `skip_cache = true` in all cases where we're NOT subsequently calling `__wt_blkcache_put()`.

Please let me know if that's wrong.

Reply:
Ah, I see. Your reasoning is correct.  If the variable were named skip_cache_put, the reasoning would be more intuitive from code. Would you be opposed to renaming it? (I know this makes the name longer).

Reply:
Agreed, fixed in eed1be7fc.